env0/resource_project_policy.go

```diff
@@ -92,9 +111,14 @@ func setPolicySchema(d *schema.ResourceData, policy client.Policy) error {
 }
 
 func resourcePolicyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := resourcePolicyUpdate(ctx, d, meta); err != nil {
```

Comment:
Should setId only if the update was successful (this was a bug I found during testing).